pkg/auth/audit/audit.go

```diff
@@ -208,3 +212,56 @@ func isExist(array []string, key string) bool {
 	}
 	return false
 }
+
+func (a *auditLog) concealSensitiveData(requestURI string, body []byte) []byte {
+	var m map[string]interface{}
+	if err := json.Unmarshal(body, &m); err != nil {
+		return body
+	}
+
+	var changed bool
+	if strings.HasSuffix(requestURI, "secrets") {
+		// Conceal values of secret data.
+		dataKey := "data"
+		data, _ := m[dataKey].(map[string]interface{})
+		if data == nil {
+			dataKey = "stringData"
+			data, _ = m[dataKey].(map[string]interface{})
+		}
+
+		for key := range data {
+			data[key] = redacted
+		}
+		if data != nil {
+			changed = true
+			m[dataKey] = data
+		}
+	}
+
+	if !a.concealMap(m) && !changed {
+		return body
+	}
+
+	newBody, err := json.Marshal(m)
+	if err != nil {
+		return body
+	}
+	return newBody
+}
+
+func (a *auditLog) concealMap(m map[string]interface{}) bool {
+	var changed bool
+	for key := range m {
+		if _, ok := m[key].(string); ok {
+			if a.keysToConcealRegex.MatchString(key) {
+				changed = true
+				m[key] = redacted
+			}
+		} else if nested, ok := m[key].(map[string]interface{}); ok && a.concealMap(nested) {
```

Comment:
Now that nested maps are being accounted for, would performance be impacted? Should it be retested?

Reply:
It was actually faster: 1.599742ms 549.326µs